node/fxutil/util.go

```diff
@@ -0,0 +1,19 @@
+package fxutil
+
+import (
+	"context"
+
+	"go.uber.org/fx"
+)
+
+// WithLifecycle wraps a context to be canceled when the lifecycle stops.
+func WithLifecycle(ctx context.Context, lc fx.Lifecycle) context.Context {
+	ctx, cancel := context.WithCancel(ctx)
+	lc.Append(fx.Hook{
+		OnStop: func(_ context.Context) error {
+			cancel()
+			return nil
+		},
+	})
+	return ctx
+}
```

Comment:
Maybe I'm too critical with git but this should really be recognized as a moved file IMO.

Reply:
Yeah